dra-evolution/pkg/api/claim_types.go

```diff
@@ -1,379 +1,607 @@
 package api
 
 import (
-	"k8s.io/apimachinery/pkg/api/resource"
+	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
 )
 
 // DeviceClass is a vendor or admin-provided resource that contains
-// contraint and configuration information. Essentially, it is a re-usable
-// collection of predefined data that device claims may use.
+// device configuration and requirements. It can be referenced in
+// the device requests of a claim to apply these presets.
 // Cluster scoped.
 type DeviceClass struct {
-	metav1.TypeMeta   `json:",inline"`
-	metav1.ObjectMeta `json:"metadata,omitempty"`
-
-	Spec DeviceClassSpec `json:"spec,omitempty"`
-}
+	metav1.TypeMeta `json:",inline"`
+	// Standard object metadata
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
 
-// DeviceClassSpec provides either a selector to find other
-// classes, or the specific constraints defining this class.
-type DeviceClassSpec struct {
+	// Only nodes matching the selector will be considered by the scheduler
+	// when trying to find a Node that fits a Pod when that Pod uses
+	// a ResourceClaim that has not been allocated yet.
+	//
+	// Setting this field is optional. If unset, all nodes are candidates.
+	// It is ignored when the claim does not use a control plane controller
+	// for allocation.
+	//
+	// This is an alpha field and requires enabling the DRAControlPlaneController
+	// feature gate.
+	//
+	// +optional
+	SuitableNodes *v1.NodeSelector `json:"suitableNodes,omitempty" protobuf:"bytes,4,opt,name=suitableNodes"`
 
-	// One of selector and classCriteria must be populated.
+	// Config defines configuration parameters that apply to each request in a claim using this class.
+	// They are ignored while allocating the claim.
+	//
+	// +optional
+	// +listType=atomic
+	Config []ConfigurationParameters `json:"config,omitempty"`
```

Comment:
What's the use-case for a list?

Reply:
The filter may match devices from different vendors ("give me a NIC, not matter from which vendor"). Then user may still want the ability to configure the actual device ("Give me a NIC, no matter from which vendor. If its from vendor A, use this config, If its from vendor B, use this instead.").

Reply:
A device from vendor C then wouldn't get explicit configuration. If users don't want that, they can filter for devices from "known" vendors.

Reply:
I see.  That seems plausible but "advanced".

Reply:
Can resolve this, will add a comment suggestion above